controller/scan/registry.go

```diff
@@ -879,51 +879,114 @@ func (rs *Registry) checkAndPutImageResult(sctx *scanContext, id string, result
 		// not canceled, continue
 	}
 
+	if result.ScanTypesRequested == nil {
```

Comment:
When result.ScanTypesRequested is nil, does it mean scanner is older version that supports vulnerability scan only?
If so, should 'Vulnerability' filed be set to true when result.ScanTypesRequested is nil ?

Reply:
Ok. This has been implemented. Scanner has also been updated so that the newer versions never return a nil `req.ScanTypesRequested` so that this behavior is consistent.